### ISSUE TITLE
"CfnParseError: Duplicate resource found" error thrown regardless of configuration

### ISSUE THREAD

**Issue by diranged** [aws-cloudformation/cfn-lint#2211]
*cfn-lint version: (`cfn-lint --version`)*

cfn-lint 0.58.0

*Description of issue.*

While trying to run [TaskCat](https://github.com/aws-ia/taskcat/) against one of our bigger CFN stacks, we ran into an error where TaskCat fails with a `CfnParseError`. The error comes from [TaskCat calling](https://github.com/aws-ia/taskcat/blob/b1737311b19cf976af7a60e77f752c968ea8cf6b/taskcat/_cfn/template.py#L28) cfn-lint at https://github.com/aws-cloudformation/cfn-lint/blob/v0.58.0/src/cfnlint/decode/cfn_yaml.py#L83-L87 to load up the CFN template.

In our case, the situation is that we have a bunch of nearly identical resources in a template. We use YAML Anchors to replicate all of the properties ... but then we override two or three properties on each of the replicated resources. Eg:

```yaml
  DatabaseInstance0:
    Type: AWS::RDS::DBInstance
    Properties: &DefaultInstanceProperties
      AllowMajorVersionUpgrade: True
      AutoMinorVersionUpgrade: {Ref: EnableAutoMinorVersionUpgrade }
      DBParameterGroupName:       {Ref: DBParameterGroup}
      DBInstanceClass:            {Ref: DBInstanceClass}
      CACertificateIdentifier:
        Fn::If:
          - HasCACertificateIdentifier
          - Ref: CACertificateIdentifier
          - Ref: 'AWS::NoValue'
      Engine: aurora-postgresql
      DBClusterIdentifier: {Ref: DatabaseCluster}
      DBSubnetGroupName: {Ref: DBSubnetGroup}
...

  DatabaseInstance1:
    Type: AWS::RDS::DBInstance
    DependsOn: DatabaseInstance0
    Properties:
      <<: *DefaultInstanceProperties
      PromotionTier:
        Fn::If: [SetPromotionTier, 1, Ref: 'AWS::NoValue']
      DBInstanceClass: {Ref: SecondaryDBInstanceClass}
      DBInstanceIdentifier:
        Fn::Join: ['-', [Ref: DatabaseCluster, 1, Ref: Instance1Zone]]
      AvailabilityZone:
        Fn::Join: ['', [Ref: 'AWS::Region', Ref: Instance1Zone]]
```

Unfortunately because of the hard-coded cfn-lint check for duplicates _at load time_, we are unable to skip this particular check and taskcat fails.

```
[ERROR  ] : CfnParseError Duplicate resource found "PromotionTier" (line 1234)
Traceback (most recent call last):
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/taskcat/_cli.py", line 46, in main 
    cli.run()
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/taskcat/_cli_core.py", line 314, in run
    return getattr(command(), subcommand)(**args)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/taskcat/_cli_modules/test.py", line 167, in run
    with test:
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/taskcat/testing/base_test.py", line 58, in __enter__
    raise ex
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/taskcat/testing/base_test.py", line 55, in __enter__
    self.run()
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/taskcat/testing/_cfn_test.py", line 83, in run
    templates = self.config.get_templates()
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/taskcat/_config.py", line 411, in get_templates
    templates[test_name] = Template(
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/taskcat/_cfn/template.py", line 59, in __init__
    self.template = self.template_cache.get(str(self.template_path))
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/taskcat/_cfn/template.py", line 28, in get
    self._templates[template_path] = cfnlint.decode.cfn_yaml.load(
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/cfnlint/decode/cfn_yaml.py", line 246, in load 
    return loads(content, filename)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/cfnlint/decode/cfn_yaml.py", line 224, in loads
    template = loader.get_single_data()
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/yaml/constructor.py", line 51, in get_single_data
    return self.construct_document(node)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/yaml/constructor.py", line 55, in construct_document
    data = self.construct_object(node)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/cfnlint/decode/cfn_yaml.py", line 81, in construct_yaml_map
    value = self.construct_object(value_node, False)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/cfnlint/decode/cfn_yaml.py", line 81, in construct_yaml_map
    value = self.construct_object(value_node, False)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/cfnlint/decode/cfn_yaml.py", line 81, in construct_yaml_map
    value = self.construct_object(value_node, False)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/Users/foo/git/org/repo/.venv/lib/python3.9/site-packages/cfnlint/decode/cfn_yaml.py", line 85, in construct_yaml_map
    raise CfnParseError(
cfnlint.decode.cfn_yaml.CfnParseError: Duplicate resource found "PromotionTier" (line 1234)

```
